### PR TITLE
Upgrade to newer Dokka and Knit

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: reactive/kotlinx-coroutines-rx2/build.gradle [hunk 1]
```diff
@@ -9,10 +10,12 @@ dependencies {
...
-tasks.withType(dokka.getClass()) {
-    externalDocumentationLink {
-        url = new URL('http://reactivex.io/RxJava/2.x/javadoc/')
-        packageListUrl = projectDir.toPath().resolve("package.list").toUri().toURL()
+tasks.withType(DokkaTaskPartial.class) {
+    dokkaSourceSets.configureEach {
+        externalDocumentationLink {
+            url.set(new URL('http://reactivex.io/RxJava/2.x/javadoc/'))
+            packageListUrl.set(projectDir.toPath().resolve("package.list").toUri().toURL())
+        }
```
Comment (dkhalanskyjb) [inline reactive/kotlinx-coroutines-rx2/build.gradle:18]: Looking through `git grep externalDoc`, I notice the lack of Swing and JavaFX modules. It looks like our documentation doesn't reference anything Swing-related, so such a directive is not needed there, but the same is not true for JavaFX.

Also, why does this look different from the way it's done in the [reactive module](https://github.com/Kotlin/kotlinx.coroutines/blob/b852b20131fff5c0d605d7cb5f8001b715f02c1e/reactive/kotlinx-coroutines-reactive/build.gradle.kts#L34-L36)? It looks much more succinct there, and I don't understand the difference between these two modules in this regard.

**Review by dkhalanskyjb — CHANGES_REQUESTED** [review]
Can't comment on the changes to the URLs, as I don't understand them. Also, I have only a surface-level understanding of the changes to the build scripts. Hope that's enough.

File: buildSrc/src/main/kotlin/Dokka.kt [hunk 2]
```diff
@@ -17,10 +15,20 @@ fun Project.externalDocumentationLink(
     url: String,
     packageList: File = projectDir.resolve("package.list")
 ) {
+    tasks.withType<DokkaTaskPartial>().configureEach {
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply buildSrc/src/main/kotlin/Dokka.kt:18]: Thanks

File: gradle.properties [hunk 3]
```diff
@@ -53,6 +53,6 @@ jekyll_version=4.0
 
 # JS IR backend sometimes crashes with out-of-memory
 # TODO: Remove once KT-37187 is fixed
-org.gradle.jvmargs=-Xmx2g
+org.gradle.jvmargs=-Xmx4g
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply gradle.properties:56]: Dokka is quite memory heavy, so yes

File: reactive/kotlinx-coroutines-rx2/build.gradle (hunk 1 above)
Comment (qwwdfsad) [reply reactive/kotlinx-coroutines-rx2/build.gradle:18]: FTR: discussed that reactive module leverages our kts extensions

Regarding JavaFX -- seems to be a general (not PR-related oversight) that should be fixed

**Pull request merged**